src/main/sensors/gyro_init.c

```diff
@@ -603,68 +592,69 @@ bool gyroInit(void)
         eepromWriteRequired = true;
     }
 
-#if defined(USE_MULTI_GYRO)
-    if ((gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_BOTH && !((gyroDetectionFlags & GYRO_ALL_MASK) == GYRO_ALL_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_1 && !(gyroDetectionFlags & GYRO_1_MASK))
-        || (gyro.gyroToUse == GYRO_CONFIG_USE_GYRO_2 && !(gyroDetectionFlags & GYRO_2_MASK))) {
-        if (gyroDetectionFlags & GYRO_1_MASK) {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_1;
-        } else {
-            gyro.gyroToUse = GYRO_CONFIG_USE_GYRO_2;
+    // check if all enabled sensors are detected
+    for (int i = 0; i < GYRO_COUNT; i++) {
```

Comment:
(this is `gyroDetectionFlags & gyroConfig()->gyro_enabled_bitmask` )

Reply:
Won't this give true even if only some of the detected and enabled gyros are the same?

Reply:
wait, no you are right